connectivity/http/http.c

```diff
@@ -256,14 +251,18 @@ static int ta_http_process_request(ta_http_t *const http, iota_client_service_t
     return process_options_request(out);
   }
 
-  if (api_path_matcher(url, "/mam/[A-Z9]{81}[/]?") == SC_OK) {
-    return process_recv_mam_msg_request(http, iota_service, url, out);
-  } else if (api_path_matcher(url, "/mam[/]?") == SC_OK) {
-    if (payload != NULL) {
-      return process_send_mam_msg_request(http, iota_service, payload, out);
+  if (api_path_matcher(url, "/mam/(recv|send)[/]?") == SC_OK) {
+    if (api_path_matcher(url, ".*/recv.*") == SC_OK) {
+      if (payload != NULL) {
+        return process_recv_mam_msg_request(http, iota_service, payload, out);
+      }
+      return process_method_not_allowed_request(out);
+    } else if (api_path_matcher(url, ".*send.*") == SC_OK) {
```

Comment:
`else if` statement is not necessary since you already check via regular expression.